App/Sagas/TextileSagas.js

```diff
@@ -71,59 +71,79 @@ export function * recoverPassword ({data}) {
+  } else if (newState === 'inactive') {
+    yield * triggerStopNode()
   }
+  yield call(BackgroundTimer.stop)
 }
 
-export function * startNode () {
+export function * startNode ({path}) {
+  const onboarded = yield select(TextileSelectors.onboarded)
+  if (!onboarded) {
+    return
+  }
+  yield call(BackgroundTimer.start)
   try {
-    const success = yield call(IPFS.startNode)
-    if (success) {
-      const value = IPFS.getGatewayPassword()
-      if (Platform.OS === 'android') {
-        CookieManager.setFromResponse(
-          'https://localhost:9080',
-          'SessionId=' + value + '; path=/; expires=Thu, 1 Jan 2030 00:00:00 -0000; secure; HttpOnly')
-            .then((res) => {
-              console.log('CookieManager.setFromResponse =>', res);
-            });
+    const createNodeSuccess = yield call(IPFS.createNodeWithDataDir, path, API_URL)
+    if (createNodeSuccess) {
+      const startNodeSuccess = yield call(IPFS.startNode)
+      if (startNodeSuccess) {
+        const value = yield call(IPFS.getGatewayPassword)
+        if (Platform.OS === 'android') {
+          yield call(
+            CookieManager.setFromResponse,
+            'https://localhost:9080',
+            'SessionId=' + value + '; path=/; expires=Thu, 1 Jan 2030 00:00:00 -0000; secure; HttpOnly'
+          )
+        } else {
+          yield call(
+            CookieManager.set, {
+              name: 'SessionId',
+              value: value || 'null',
+              origin: 'https://localhost:9080',
+              domain: '.localhost',
+              version: '1',
+              path: '/',
+              expiration: '2030-01-01T00:00:00.00-00:00'
+            })
+        }
+        yield put(IpfsNodeActions.startNodeSuccess())
+        yield put(IpfsNodeActions.getPhotoHashesRequest('default'))
+        yield put(IpfsNodeActions.getPhotoHashesRequest('beta'))
       } else {
-        CookieManager.set({
-          name: 'SessionId',
-          value: value ? value : "null",
-          origin: 'https://localhost:9080',
-          domain: '.localhost',
-          version: '1',
-          path: '/',
-          expiration: '2030-01-01T00:00:00.00-00:00'
-        })
+        yield put(IpfsNodeActions.startNodeFailure(new Error('Failed starting node, but no error was thrown - Should not happen')))
       }
-      yield put(IpfsNodeActions.startNodeSuccess())
     } else {
-      yield put(IpfsNodeActions.startNodeFailure(new Error('Failed starting node, but no error was thrown - Should not happen')))
+      yield put(IpfsNodeActions.startNodeFailure(new Error('Failed creating node, but no error was thrown - Should not happen')))
     }
   } catch (error) {
     yield put(IpfsNodeActions.startNodeFailure(error))
+  } finally {
+    yield call(BackgroundTimer.stop)
   }
 }
 
-export function * stopNode ({newState}) {
-  if (newState !== 'inactive') {
-    return
-  }
+export function * stopNode () {
+  yield call(BackgroundTimer.start)
```

Comment:
just checking, can this be scheduled by the OS to happen at some later date, or does it necessarily happen right away? if the former case is true, hopefully it's canceled before the node starts again

Reply:
man, good question. it's on my list to understand how to compose all these flows in a way where things are canceled etc as needed, but that's some complicated stuff. this should happen right away, but it seems possible that the app could be foregrounded again or receive a location update before the node is done stopping. It's a complicated dependency graph of things

Reply:
yeah, tricky stuff. good thing the node shuts down almost immediately